routing_deep_dive.prolific

```diff
@@ -0,0 +1,130 @@
+    ```
+    cat /var/vcap/jobs/route_emitter/config/route_emitter.json | jq . | grep nats
+    ```
+0. Use the nats gem to connect to nats:
+```nats-sub "*.*" -s nats://NATS_USERNAME:NATS_PASSWORD@NATS_ADDRESS```
+
+ The Route Emitter registers routes every 20 seconds (by default) so that the GoRouter (which subscribes to these messages) has the most up-to-date information about which IPs map to which apps and routes. Depending on how many routes there are, this might be a lot of information.
+
+0. Make a new route for one of your apps and find the route registration message.
+
+### Expected Result
+Inspect NATs messages. Look at what route information is sent to the GoRouter.
+
+## Resources
+[NATS message bus repo](https://github.com/nats-io/gnatsd)
+[NATS ruby gem repo](https://github.com/nats-io/ruby-nats)
+
+
+L: routing
+---
+
+Look at the GoRouter Routes Table
+
+### What?
+So the Route Emitter emits routes via the NATS message Bus. GoRouter _subscribes_ to those messages and keeps a route table that is uses to route network traffic bound for CF apps and CF components.
+
+Let's take a look at that route table.
+
+### How?
+0. Bosh ssh onto the router vm and become root.
+0. Install jq (a json manipulation and display tool)
+ ```
+ apt-get install jq
+ ```
+0. Get the username and password for the routing api
+ ```
+ head /var/vcap/jobs/gorouter/config/gorouter.yml
+ ```
+0. Get the routes table
+ ```
+ curl "http://USERNAME:PASSWORD@localhost:8080/routes" | jq .
+ ```
+0. Make a new route for one of your apps and find the route in the routes table.
+
+ It should look something like this (I took out some extra bits):
+    ```
+    "proxy.meow.cloche.c2c.cf-app.com": [   <------ The name of the route!
+        {
+          "address": "10.0.1.12:61014",     <------ This is the Diego Cell IP and port where GoRouter will send traffic for this route
+          "tls": true                       <------ This means the GoRouter will use send traffic to this app over TLS
+        }
+      ]
+    ```
+
+0. Check the environment variables for your app and see that the information matches the routing table
+ ```
+cf ssh MY-APP -c "env | grep CF_INSTANCE_ADDR"
+ ```
+
+### Expected Result
+You are able to access the route table on the router VM.
+
+## Resources
+[GoRouter routing table docs](https://github.com/cloudfoundry/gorouter#the-routing-table)
+
+L: routing
+---
+
+Interested in learning more about routing?
+
+🤓 If you are interested in learning even more about routing and networking, check out these [CF Networking Onboarding stories](https://github.com/pivotal/cf-networking-program-onboarding).
+
+🐞 These stories are intended to help engineers to learn how container networking and routing components work under-the-hood and how to debug them.
+
+➕ To add these stories to this tracker: clone the [repo](https://github.com/pivotal/cf-networking-program-onboarding), run the build script, and then import the created CSV to your tracker.
```

Comment:
For some reason my brain wants 

>into your tracker

rather than

>to your tracker

Not clear to me if there are actual grammar rules here of if it's mostly subjective interpretation. 🤷‍♂️